pipelinewise/cli/pipelinewise.py

```
@@ -1621,6 +1655,141 @@ def encrypt_string(self):
+        target_config = self.target['files']['config']
+
+        self.drop_pg_slot = False
+
+        cons_target_config = self.create_consumable_target_config(
+            target_config, tap_inheritable_config
+        )
+
+        # Output will be redirected into target and tap specific log directory
+        log_dir = self.get_tap_log_dir(target_id, tap_id)
+        current_time = datetime.utcnow().strftime('%Y%m%d_%H%M%S')
+
+        try:
+            with pidfile.PIDFile(self.tap['files']['pidfile']):
+                self.tap_run_log_file = os.path.join(
+                    log_dir, f'{target_id}-{tap_id}-{current_time}.partialsync.log'
+                )
+
+                # Create parameters as NamedTuples
+                tap_params = TapParams(
+                    tap_id=tap_id,
+                    type=tap_type,
+                    bin=self.tap_bin,
+                    python_bin=self.tap_python_bin,
+                    config=tap_config,
+                    properties=tap_properties,
+                    state=tap_state,
+                )
+
+                target_params = TargetParams(
+                    target_id=target_id,
+                    type=target_type,
+                    bin=self.target_bin,
+                    python_bin=self.target_python_bin,
+                    config=cons_target_config,
+                )
+
+                transform_params = TransformParams(
+                    bin=self.transform_field_bin,
+                    config=tap_transformation,
+                    python_bin=self.transform_field_python_bin,
+                    tap_id=tap_id,
+                    target_id=target_id,
+                )
+
+                self.run_tap_partialsync(
+                    tap=tap_params, target=target_params, transform=transform_params,
+                )
+
+        except pidfile.AlreadyRunningError:
+            self.logger.error('Another instance of the tap is already running.')
+            sys.exit(1)
```

Please raise an exception

---

- [ ] Fix it!

---

Yes I know, please re-raise it, the program will exit with non-zero code anyway and we will get better error logs than just `SystemExit`

---

- [ ] Raise exception

---

The other CLI commands - `run_tap`, `stop_tap`, `sync_tables` - they all exit with `sys.exit(1)` in case of a known error. Here, it's easy to figure out that the tap is already running. So better to keep it consistent and not raise an exception.